rust-runtime/aws-smithy-http/src/byte_stream/error.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::error::Error as StdError;
+use std::fmt;
+use std::io::{Error as IoError, ErrorKind as IoErrorKind};
+
+#[derive(Debug)]
+pub(super) enum ErrorKind {
+    #[cfg(feature = "rt-tokio")]
+    OffsetLargerThanFileSize,
+    #[cfg(feature = "rt-tokio")]
+    LengthLargerThanFileSizeMinusReadOffset,
```

Comment:
I didn't even realize you could have feature-gated variants like this, very neat